### DISCUSSION TITLE
Custom physical surface resolution with winit

### DISCUSSION THREAD

**Discussion by PPakalns** [gfx-rs/wgpu#3716] [Q&A]
A lot of games allow to set custom resolution that doesn't match with physical window size in pixels for windowed or fullscreen windows.

I tried to investigate, but couldn't find any example or api that provides such functionality.

Is it implemented with rendering to offscreen textures and rendering them onto full physical size textures or is there a simpler way.

Thanks for the help!

**Comment by kpreid** [comment]
> Is it implemented with rendering to offscreen textures and rendering them onto full physical size textures 

Yes.

Note that implementing it yourself isn't _just_ boilerplate; you get to control the upsampling used (standard linear texture filtering might not be the most pleasant choice), and other screen-space effects (tone-mapping, FXAA, etc) may also be done at this point.
